File: semester-2/programming/labs/zad8_battleship_game/Ships/ShipManager.cpp
//
// Created by marling on 6/7/23.
//

#include "ShipManager.hpp"

std::vector<std::shared_ptr<Ship>> ShipManager::createShips() {
    std::vector<std::shared_ptr<Ship>> ships;
    Coordinates coord(-1, -1);
    ships.emplace_back(std::make_shared<Carrier>(0, coord));
    ships.emplace_back(std::make_shared<Battleship>(0, coord));
    ships.emplace_back(std::make_shared<Cruiser>(0, coord));
    ships.emplace_back(std::make_shared<Destroyer>(0, coord));
    ships.emplace_back(std::make_shared<Destroyer>(0, coord));
    ships.emplace_back(std::make_shared<Submarine>(0, coord));
    ships.emplace_back(std::make_shared<Submarine>(0, coord));
    ships.emplace_back(std::make_shared<Submarine>(0, coord));
    return ships;
}

File: semester-3/Python/qt_battleship_game/game/ships/ship_manager.py
from .ship import Carrier, Battleship, Cruiser, Destroyer, Submarine


class ShipManager:
    def __init__(self):
        self._ships = {"submarine-1": Submarine(),
                       "submarine-2": Submarine(),
                       "submarine-3": Submarine(),
                       "destroyer-1": Destroyer(),
                       "destroyer-2": Destroyer(),
                       "cruiser-1": Cruiser(),
                       "battleship-1": Battleship(),
                       "carrier-1": Carrier()}

    def __iter__(self):
        return iter(self._ships)

    def get_ship(self, name):
        return self._ships[name]

    @property
    def ships(self):
        return self._ships